[READY] Giant Spider balance adjustments and fixes (updated)

BlueNexus opened the pull request:
<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->
- Increases the amount of toxin injected by hunters, in line with their intended purpose according to the code. Removed their chance to inject extra toxin though.
- Decreased the amount of toxin injected by guards and nurses.
- Fixed nurse spiders being unable to implant eggs into people.
- Moves the chance to inject extra stuff over into a new var.

This will be squashed upon request.

BlueNexus committed:
spider venom adjustments and fixes

Also fixed nurse spider egg implantation

mustafakalash commented:
Why so many spider buffs? Spiders isn't a game monde.

BlueNexus commented:
@mustafakalash Firstly, there hasn't been anything affecting giant spiders balance-wise in a long time.

And, Because toxin is literally the weakest poison in the game and it makes no sense for spiders to be injecting it.

BlueNexus commented:
Adjusted the values: Hunters now inject less venom.

PsiOmegaDelta commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -72,16 +75,16 @@
 				L.reagents.add_reagent(poison_type, 5)
 
 /mob/living/simple_animal/hostile/giant_spider/nurse/AttackingTarget()
 	var/target = ..()
```
Remove this target reassignment to make the below run properly every time.
Still call ..() tho.

BlueNexus replied:
Resolved

PsiOmegaDelta commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -72,16 +75,16 @@
-			if(prob(poison_per_bite))
+			L.reagents.add_reagent("venom", poison_per_bite)
+			if(prob(poison_extra_chance))
 				L << "\red You feel a tiny prick."
```
Change this to `<span class='warning'>` while you're at it.

BlueNexus replied:
Resolved.

BlueNexus commented:
Squashed and ready to go.

PsiOmegaDelta commented on code/modules/mob/living/simple_animal/hostile/giant_spider.dm:
```diff
@@ -72,16 +75,15 @@
 	if(ishuman(target))
 		var/mob/living/carbon/human/H = target
-		if(prob(poison_per_bite))
+		if(prob(poison_extra_chance))
```
Ah, had to unroll the diff. Cannot comment where applicable..
Line 74,75,76: Replace target with .
Above line 84: Add: . = .. ()
Line 84, 85: Replace target with .

BlueNexus replied:
Resolved, hopefully.

BlueNexus commented:
Passed travis.

BlueNexus commented:
Alright, this is ready to go once approved.

comma commented:
> While their venom is stronger, they inject less venom overall.

okey
Nurses used to inject 10 units of toxin, then 10% chance to add 10 sleeptox/eggs
Now it's 5 units of venom (which is twice the power), with 5% chance of sleeptox/eggs
So overall, same amount of damage. 10 units of power 4 toxin vs 5 units of power 8 toxins. Tomato, tomato.

Now hunters:
Used to be: 5 units of toxin, 5% chance of 5 units more.
Now: 10 guaranteed units of venom.
Overall, 2-4 times the damage.

Yes this is an outright buff.

BlueNexus commented:
@comma Yes, of course giving them some stronger venom is going to be a small buff. It's either that or leave them with Toxin, which is the weakest poison in the game, and it makes no sense that these giant spiders are injecting almost harmless toxin. 

If people want me to remove the new venom, I can just go back to using toxin and just increase how much Hunters inject, in line with their comments.

comma commented:
I don't really care either way, just didn't like confusing 'this is not a buff, everything is same' when actually it's increasing damage up to 4 times.

BlueNexus commented:
Reworking this a little.

BlueNexus commented:
Please keep in mind that as Techhead said, Dylovene can still easily counter this. Yes, it's a bit of a buff, and I've removed everything stating otherwise. I've adjusted the values after a discussion on IRC, so we should be good now.

BlueNexus commented:
Ready for review again, with the adjusted values. Hopefully people should be a bit more happy with this.

BlueNexus commented:
If whoever reviews this is unhappy with the venom changes, I am willing to remove all the venom changes, just increase how much toxin hunters inject, and keep the fixes.

Aticius commented:
I'm horrified and disgusted by implanting eggs into people. 

But really, I'm all for making spider toxloss more important- as it stands you poke them with a anti-toxin needle, give them a pat on the back and a bandaid and they're fine. 

Venom would make this much nicer.

Kasuobes commented:
^ Not really, because toxins metabolize much slower than anti-tox, they stay around in your system for a good while.

BlueNexus commented:
As a note, if whoever reviews this is unhappy with the venom changes, I will remove the venom changes, just increase how much toxin hunters inject, and keep the fixes.

Kelenius commented:
Dylovene DOES NOT neutralize toxins anymore. It only removes damage. It doesn't touch reagents.

mustafakalash commented:
> it makes no sense that these giant spiders are injecting almost harmless toxin

Realism isn't a good reason for balance changes. Yes, random events shouldn't have zero effect on the round, but they certainly shouldn't be round consuming. As they stand, I don't think spiders need all these buffs. Even if this isn't a big buff, you've been doing a lot of "not big buffs" to spiders.

BlueNexus commented:
@mustafakalash I haven't been doing any "not big buffs" to spiders. I closed all my other spider PRs.

BlueNexus commented:
If someone could just tell me what to change to make this PR acceptable, that would be really great.

BlueNexus committed:
Balance tweaks

BlueNexus commented:
Lowered the amount of venom injected again.

BlueNexus committed:
Reverted the new venom
But left hunters with a little more toxin in line with their comments.

BlueNexus commented:
Reverted the new venom.
If people would like the toxin amounts to be adjusted, please say so and I will do so.

BlueNexus committed:
Moar balance adjustments

BlueNexus committed:
Changelog
